loaders/authentication.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch');
+const dotenv = require('dotenv').config();
+
+//base URL of cigarra authentication
+const URL = 'https://sigarra.up.pt/feup/pt/mob_val_geral.autentica?'
+
+//build URL with credentials
+let urlRequest = URL + 'pv_login=' + process.env.UP_USER + '&pv_password=' + process.env.UP_PWD;
+
+let cookie = null;
+
+//Called sigarra api to make authentication
+async function updateCookie(){
+   try {
+        const response = await fetch(urlRequest, {
+            method: 'POST'
+        });
+        if(response.ok){
+            return await response.json();
+        }
+        throw new Error('Request Failed: ' + response.status + "-" + response.statusText);
+
+   }catch (error){
+       console.log(error);
+   }
+}
+
+//give value to cookie
+async function getCookie(){
```

Comment:
The method names are switched... `getCookie` updates the cookie variable and `updateCookie` only requests it. I recommend a switch, and would re-evaluate each function's objective as well :wink: